### PR TITLE
feat: add Package Manager test to health_checks

### PR DESCRIPTION
**Pull request by 0618**
## Problem

**Issue number, if available:**

https://github.com/aws-amplify/amplify-backend/issues/961

## Changes

- merge poc-e2e-flow-test.yml into health_checks
- create re-usable action `tag-trigger-action`

**Corresponding docs PR, if applicable:**

## Validation

~~https://github.com/0618/amplify-backend/actions/runs/7733532723~~
https://github.com/0618/amplify-backend/actions/runs/7733922426?pr=1

## Checklist

- [ ] If this PR includes a functional change to the runtime behavior of the code, I have added or updated automated test coverage for this change.
- [ ] If this PR requires a change to the [Project Architecture README](../PROJECT_ARCHITECTURE.md), I have included that update in this PR.
- [ ] If this PR requires a docs update, I have linked to that docs PR above.
- [ ] If this PR modifies E2E tests, makes changes to resource provisioning, or makes SDK calls, I have run the PR checks with the `run-e2e` label set.

_By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license._

### CODE REVIEW

File: .github/workflows/health_checks.yml [hunk 1]
```diff
@@ -110,21 +110,29 @@ jobs:
...
+      GH_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+    outputs:
+      run_package_manager_tests: ${{ steps.check.outputs.run_package_manager_tests }}
+    steps:
+      - name: Check event is push to main or pull request has run-e2e label
+        id: check
+        uses: ./.github/actions/tag_trigger_action
+        with:
+          tag-name: run-package-manager-tests
+          test-name: run_package_manager_tests
```
Comment (sobolk) [inline .github/workflows/health_checks.yml:135]: I think we should just use `run-e2e` label to include package manager tests.
package manager test are a kind of e2e tests.
I.e. I'd revert this part and make `run_package_manager_tests_tests` below dependent on `run_e2e` "GITHUB_OUTPUT".

File: .github/workflows/health_checks.yml [hunk 2]
```diff
@@ -164,6 +172,53 @@ jobs:
           aws-region: us-west-2
       - name: Run E2E tests
         run: npm run e2e
+  run_package_manager_tests_tests:
```
Comment (sobolk) [inline .github/workflows/health_checks.yml:175]:
```suggestion
  run_package_manager_e2e_tests:
```

File: .github/workflows/health_checks.yml [hunk 3]
```diff
@@ -164,6 +172,53 @@ jobs:
...
+        os: [ubuntu-latest, macos-latest, windows-latest]
+        pkg-manager: [npm, yarn-classic, yarn-modern, pnpm]
+        node-version: [20]
+        exclude:
+          - os: windows-latest
+            pkg-manager: pnpm
+    env:
+      PACKAGE_MANAGER: ${{ matrix.pkg-manager }}
+    runs-on: ${{ matrix.os }}
+    timeout-minutes: 60
```
Comment (sobolk) [inline .github/workflows/health_checks.yml:190]: can timeout be smaller? how long these tests run ?

File: .github/actions/tag_trigger_action/action.yml [hunk 4]
```diff
@@ -0,0 +1,26 @@
+name: tag_trigger_action
+description: Trigger a workflow on a tag addition
```
Comment (edwardfoyle) [inline .github/actions/tag_trigger_action/action.yml:2]: I wonder if this should be called something like `job_execution_gate`? It's doing more than just checking a tag value.

Also, it's specifically setting the "run_e2e" output which seems too specific if we're making this a reusable action. Maybe it should just output "continue_execution=true|false"?

File: .github/actions/tag_trigger_action/action.yml [hunk 5]
```diff
@@ -0,0 +1,26 @@
...
+inputs:
+  tag-name:
+    description: 'The tag to trigger the workflow on'
+    required: true
+  test-name:
```
Comment (edwardfoyle) [inline .github/actions/tag_trigger_action/action.yml:7]: I don't see where this input is used?

File: .github/actions/tag_trigger_action/action.yml (hunk 4 above)
Comment (0618) [reply .github/actions/tag_trigger_action/action.yml:2]: if we use `run-e2e` test, the re-usable action is unnecessary

File: .github/actions/tag_trigger_action/action.yml (hunk 5 above)
Comment (0618) [reply .github/actions/tag_trigger_action/action.yml:7]: it's used in `... grep ${{ inputs.tag-name }} --quiet;...` 😂

File: .github/workflows/health_checks.yml (hunk 2 above)
Comment (0618) [reply .github/workflows/health_checks.yml]: Done af8d396dfb

File: .github/workflows/health_checks.yml (hunk 1 above)
Comment (0618) [reply .github/workflows/health_checks.yml]: Done af8d396dfb

File: .github/workflows/health_checks.yml (hunk 3 above)
Comment (0618) [reply .github/workflows/health_checks.yml]: changed to 25 min af8d396dfb

**Review by sobolk — APPROVED** [review]

**Review by edwardfoyle — APPROVED** [review]

**Pull request merged**